lib/kubernetes-deploy/resource_watcher.rb

```diff
@@ -41,14 +41,19 @@ def run(delay_sync: 3.seconds, reminder_interval: 30.seconds)
     def report_what_just_happened(resources, watch_time)
       new_successes, new_failures = resources.partition(&:deploy_succeeded?)
       new_failures.each do |resource|
+        tags = %W(namespace:#{@namespace} sha:#{ENV['REVISION']} resource:#{resource.id})
         if resource.deploy_failed?
+          ::StatsD.increment('failure', 1, tags: tags)
           @logger.error("#{resource.id} failed to deploy after #{watch_time}s")
         else
+          ::StatsD.increment('timeout', 1, tags: tags)
           @logger.error("#{resource.id} deployment timed out")
         end
       end
 
       if new_successes.present?
+        tags = %W(namespace:#{@namespace} sha:#{ENV['REVISION']} resources:#{new_successes.map(&:id).join(';')})
+        ::StatsD.measure('duration.success', watch_time * 1000, tags: tags)
```

Comment:
Yeah, the higher interval is more of a hack than solution. We would need to have real watcher for this to work ideally.